pkg/api/persistentvolumeclaim/util.go

```diff
@@ -43,7 +43,9 @@ func DropDisabledFields(pvcSpec *core.PersistentVolumeClaimSpec) {
 	// feature gate is disabled and dataSourceRef.Namespace is specified.
 	if !utilfeature.DefaultFeatureGate.Enabled(features.CrossNamespaceVolumeDataSource) &&
 		pvcSpec.DataSourceRef != nil && len(pvcSpec.DataSourceRef.Namespace) != 0 {
-		pvcSpec.DataSourceRef = nil
+		if !dataSourceRefInUse(oldPVCSpec) {
```

Comment:
Can you also add this to the `AnyVolumeDataSource` check above?

Reply:
Added (https://github.com/kubernetes/kubernetes/pull/113186/commits/3f6ee41152580bbfd2c221be6c05a7619fa68b07)